packages/runtime/test/integration/pool-transaction-fee-test.ts

```diff
@@ -86,4 +98,85 @@ describe("Pooled Transaction Fees Test", function () {
 			RUNTIME_ERRORS.TRANSACTION.FEES_NOT_ENOUGH
 		);
 	});
+
+	it("Should unfunded accounts be able to issue transactions of group size 3", () => {
+		const amount = 4000;
+		const fee = 3000;
+		// group with fee distribution. Pooled transaction fee
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns.
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: bob.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: bob.account,
+				toAccountAddr: john.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+		];
+
+		runtime.executeTx(groupTx);
+
+		syncAccounts();
+		assert.equal(john.balance(), BigInt(initialBalance) - BigInt(fee));
+		assert.equal(alice.balance(), BigInt(initialBalance));
+		assert.equal(bob.balance(), BigInt(initialBalance));
+	});
+
+	it("Should unfunded accounts be able to issue transactions and opt-in", () => {
+		setupAsset();
+		const amount = 200000;
+		const fee = 3000;
+		// group with fee distribution
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
```

Comment:
I just checked this. 3 out of 4 unit test cases fails in this file. This is expected. Only txn fee is covered in pooling.

Reply:
Why this is expected? Should it work when the first account is unfunded and the second transaction covers the fee?

Reply:
We are trying to send algos. Algos are not covered. If 1st account having 0 algos then it can't send algo but will be able to issue txn if covered by other txn.

Reply:
You want to add a scenario when fee is not enough. If yes, this test is already there.

Reply:
If account is unfunded then it won't be able to send algos but will be able to issue txn if covered by other txn given that it has sufficient fund.

Reply:
I found an article: https://developer.algorand.org/articles/avm-7-new-features/?from_query=pooling

What I understood from here is that: The unfunded accounts can call Smart Contracts and interact with a dapp, while not being funded, as long as they are grouped with a fee paying transaction.

Reply:
This scenario is covered when account is unfunded but able to issue txn in group txn.
I have also covered a scenario when account is unfunded and trying to send algos.